### PR TITLE
PFW-1354 Fix slow button processing

### PR DESCRIPTION
**Pull request by gudnimg**
JIRA ticket: https://dev.prusa3d.com/browse/PFW-1354

The issue on the MK3S side:
1. There is an MMU error screen
2. Select Retry
3. The error screen is dismissed. MK3S sends Q0 MMU Query
4. MMU Query returns back the error state
5. At this point we process the button (a bit late)

We should process the button just after we dismiss the error screen and before sending another Query Q0 request.

### CODE REVIEW

**Comment by gudnimg** [comment]
This needs to be tested well. I don't see any issue with this from the MK3S perspective. But could there be an issue where the MMU buttons are pushed at random times?

My thought is when the MMU error screen is NOT present, then `CheckUserInput` should not do anything. But I'm not sure about the physical buttons on the MMU, are there times they should not do anything?

Here is an image showing the serial stream with the above fix, it looks good.
image

**Comment by vintagepc** [comment]
Physical MMU buttons won't be affected unless during a toolchange when the "printer in charge" mode is enabled.
But you're right it has an impact on that if we want to use this mode for other things.

File: Firmware/mmu2.cpp [hunk 1]
```diff
@@ -638,27 +639,22 @@ StepStatus MMU2::LogicStep() {
...
         break;
     case VersionMismatch:
         StopKeepPowered();
         ReportError(ErrorCode::VERSION_MISMATCH);
-        CheckUserInput();
         break;
     case ButtonPushed:
         lastButton = logic.Button();
         LogEchoEvent("MMU Button pushed");
-        CheckUserInput();
```
Comment (vintagepc) [inline Firmware/mmu2.cpp:661]: This one might need to be left alone otherwise the MMU button won't be registered until the next cycle, if at all.

Comment (gudnimg) [reply Firmware/mmu2.cpp:661]: I've added this back

**Comment by gudnimg** [comment]
Rebased branch onto mk3-mmu

**Comment by gudnimg** [comment]
There is currently a known issue with the MMU buttons, where the MMU reports the button to the MK3S multiple times. This seems to be unrelated to the changes in this PR. This PR mainly resolves the LCD glitches observed when using the MMU screen.

**Review by vintagepc — APPROVED** [review]

**Comment by gudnimg** [comment]
@vintagepc Its very strange that I have not seen the issue this PR was supposed to solve again. The LCD behaved well this weekend 🤔 Smells funny, maybe we should wait with merging this one? What's your opinion?

**Pull request merged**